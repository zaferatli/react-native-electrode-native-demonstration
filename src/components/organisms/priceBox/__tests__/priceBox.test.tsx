import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import PriceBox, {PriceBoxPropsType} from '../index';

const getComponent = (props: PriceBoxPropsType) => {
  return <PriceBox {...props} />;
};

it('initial render button', () => {
  renderer.create(
    getComponent({
      value: true,
    }),
  );
});

it('Primary PriceBox', () => {
  const props: PriceBoxPropsType = {
    value: true,
    title: 'YEARLY ACCESS',
    price: '$0,11/day',
    boldPrice: '$39.99/year',
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});

it('Primary PriceBox', () => {
  const props: PriceBoxPropsType = {
    value: false,
    title: 'YEARLY ACCESS',
    price: '$0,11/day',
    boldPrice: '$39.99/year',
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});
