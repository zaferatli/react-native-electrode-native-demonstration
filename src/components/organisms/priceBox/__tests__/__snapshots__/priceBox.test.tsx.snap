// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary PriceBox 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#FFF5F5FF",
      "borderColor": "#FF0000",
      "borderRadius": 10,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": undefined,
      "height": 60,
      "justifyContent": "space-between",
      "marginTop": 15,
      "opacity": 1,
      "paddingHorizontal": 20,
      "width": 361,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#717578FF",
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "500",
          "lineHeight": undefined,
        }
      }
    >
      YEARLY ACCESS
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": undefined,
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#717578FF",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "400",
            "lineHeight": undefined,
          }
        }
      >
        $0,11/day
      </Text>
      <Text
        style={
          Object {
            "color": "#17171CFF",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "500",
            "lineHeight": undefined,
            "marginLeft": 5,
          }
        }
      >
        $39.99/year
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FF0000",
        "borderColor": "#FF0000",
        "borderRadius": 24,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "column",
        "flexWrap": undefined,
        "height": 24,
        "justifyContent": "center",
        "opacity": 1,
        "width": 24,
      }
    }
  >
    <check.svg
      fill="#FFFFFF"
      height={9}
      iconType="CheckIcon"
      width={12}
    />
  </View>
</View>
`;

exports[`Primary PriceBox 2`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#FFF5F5FF",
      "borderColor": "#E7E8E9FF",
      "borderRadius": 10,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": undefined,
      "height": 60,
      "justifyContent": "space-between",
      "marginTop": 15,
      "opacity": 1,
      "paddingHorizontal": 20,
      "width": 361,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#717578FF",
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "500",
          "lineHeight": undefined,
        }
      }
    >
      YEARLY ACCESS
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": undefined,
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#717578FF",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "400",
            "lineHeight": undefined,
          }
        }
      >
        $0,11/day
      </Text>
      <Text
        style={
          Object {
            "color": "#17171CFF",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "500",
            "lineHeight": undefined,
            "marginLeft": 5,
          }
        }
      >
        $39.99/year
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#E7E8E9FF",
        "borderRadius": 24,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "column",
        "flexWrap": undefined,
        "height": 24,
        "justifyContent": "center",
        "opacity": 1,
        "width": 24,
      }
    }
  />
</View>
`;
